### PR TITLE
kubeadm: Update versions for 1.21

### PR DESCRIPTION
**Pull request by RA489**
update current/minimal supported versions.
xref https://github.com/kubernetes/kubeadm/issues/2362

### CODE REVIEW

**Comment by RA489** [comment]
@kubernetes/sig-cluster-lifecycle-pr-reviews
/kind cleanup

File: cmd/kubeadm/app/constants/constants.go [hunk 1]
```diff
@@ -472,6 +472,7 @@ var (
 		19: "3.4.13-0",
 		20: "3.4.13-0",
 		21: "3.4.13-0",
+		22: "3.4.14-0",
```
Comment (neolit123) [inline cmd/kubeadm/app/constants/constants.go:475]:
```suggestion
		22: "3.4.13-0",
```

let's keep this version the same until the new etcd image is available.

**Comment by neolit123** [comment]
the CI is failing here and kubeadm is receiving version 1.19 for some reason?

let's see it fails again:
/retest

File: cmd/kubeadm/app/constants/constants.go (hunk 1 above)
Comment (RA489) [reply cmd/kubeadm/app/constants/constants.go]: @neolit123 updated.

**Comment by neolit123** [comment]
@RA489 

looks like some previous commits broke our unit tests a little, making them depend on hardcoded version strings:
you can apply this diff with `git apply patch.diff` and it should work.

```diff
diff --git a/cmd/kubeadm/app/cmd/alpha/kubeconfig_test.go b/cmd/kubeadm/app/cmd/alpha/kubeconfig_test.go
index 132ca498c90..b96f5147835 100644
--- a/cmd/kubeadm/app/cmd/alpha/kubeconfig_test.go
+++ b/cmd/kubeadm/app/cmd/alpha/kubeconfig_test.go
@@ -54,7 +54,7 @@ func generateTestKubeadmConfig(dir, id, certDir, clusterName string) (string, er
 		},
 		CertificatesDir:   certDir,
 		ClusterName:       clusterName,
-		KubernetesVersion: "v1.19.0",
+		KubernetesVersion: kubeadmconstants.MinimumControlPlaneVersion.String(),
 	}
 
 	var buf bytes.Buffer
diff --git a/cmd/kubeadm/app/cmd/certs_test.go b/cmd/kubeadm/app/cmd/certs_test.go
index 17979873950..a3451c269a2 100644
--- a/cmd/kubeadm/app/cmd/certs_test.go
+++ b/cmd/kubeadm/app/cmd/certs_test.go
@@ -330,7 +330,7 @@ func TestRunGenCSR(t *testing.T) {
 					ServiceSubnet: "192.0.2.0/24",
 				},
 				CertificatesDir:   certDir,
-				KubernetesVersion: "v1.19.0",
+				KubernetesVersion: kubeadmconstants.MinimumControlPlaneVersion.String(),
 			},
 		},
 	}
@@ -378,7 +378,7 @@ func TestGenCSRConfig(t *testing.T) {
 
 	// A minimal kubeadm config with just enough values to avoid triggering
 	// auto-detection of config values at runtime.
-	const kubeadmConfig = `
+	var kubeadmConfig = `
 apiVersion: kubeadm.k8s.io/v1beta2
 kind: InitConfiguration
 localAPIEndpoint:
@@ -389,8 +389,7 @@ nodeRegistration:
 apiVersion: kubeadm.k8s.io/v1beta2
 kind: ClusterConfiguration
 certificatesDir: /custom/config/certificates-dir
-kubernetesVersion: v1.19.0
-`
+kubernetesVersion: ` + kubeadmconstants.MinimumControlPlaneVersion.String()
 
 	tmpDir := testutil.SetupTempDir(t)
 	defer os.RemoveAll(tmpDir)
```

**Comment by RA489** [comment]
@neolit123 Applied the patch diff PTAL.

**Pull request merged**